### ISSUE TITLE
Injecting into middleware and param converters

### ISSUE THREAD

**Issue by felixfbecker** [PHP-DI/Silex-Bridge#15]
I would like to inject stuff into route middleware, like so:

``` php
$app->before(function (Request $request, DBConnection $conn) {
    // do stuff
});
```

but it doesn't work, instead I get passed the Application, resulting in a type hint error. It seems like Injection is currently only done for controllers, which is unfortunate. It is just as useful for middleware and param converters:

``` php
$controller->convert('user', function ($user, DBConnection $conn) {
    return User::findById($user, $conn);
});
```

**Comment by mnapoli** [comment]
Agreed this would be great to have, it would also mirror how the [Slim bridge](https://github.com/PHP-DI/Slim-Bridge) works, pull requests are welcome.

**Comment by felixfbecker** [comment]
@mnapoli I would like to prepare a PR, but I don't know where to start. Do you know where middleware and param converters get called?

**Comment by mnapoli** [comment]
@felixfbecker I honestly have no idea :) I've never customized that part of Silex.

After giving it a quick look it _seems_ middlewares might be called by this event listener: https://github.com/silexphp/Silex/blob/1.3/src/Silex/EventListener/MiddlewareListener.php#L79

**Comment by felixfbecker** [comment]
@mnapoli I figured that the magic happens in that class aswell and made first steps to overwrite it. Only need to find out now how to do the reflection

**Comment by mnapoli** [comment]
Closed by #16
